pkg/llm-d-inference-sim/response.go

```diff
@@ -283,12 +284,13 @@ func (req chatCompletionRequest) createToolCalls() ([]toolCall, string, error) {
 		}
 		calls = append(calls, call)
 	}
-	return calls, toolsFinishReason, nil
+
+	return calls, toolsFinishReason, countTokensForToolCalls(calls), nil
```

Comment:
Since in the case of without tool calls you calculate the number of tokens in the calling function and not in createResponseText, I think you should do the same here. The call to countTokensForToolsCalls should be in the calling function.

Also, I don't see the definition of countTokensForToolCalls which I assume is new code

Reply:
Moving the calculation to the caller will add another if statement there. I think it's more clear my way, we calculate the tokens if we generate the calls